src/main/scala/diplomacy/Nodes.scala

```diff
@@ -138,80 +344,296 @@ trait FormatNode[I <: FormatEdge, O <: FormatEdge] extends BaseNode {
+    *
+    * @param h source node also without sink handle.
+    * @return a [[OutwardNodeHandle]] with this node as outwardNode.
+    */
   override def :=* [EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): OutwardNodeHandle[DO, UO, EO, BO] = { bind(h, BIND_QUERY); this }
+
+  /** Connects a full node with an output => an output.
+    *
+    * <---< :*=* <---x  ==  <---x
+    * [[BIND_FLEX]] this node as sink, [[BIND_FLEX]] that node as source.
+    *
+    * @param h source node also without sink handle.
+    * @return a [[OutwardNodeHandle]] with this node as outwardNode.
+    */
   override def :*=*[EY](h: OutwardNodeHandle[DI, UI, EY, BI])(implicit p: Parameters, sourceInfo: SourceInfo): OutwardNodeHandle[DO, UO, EO, BO] = { bind(h, BIND_FLEX);  this }
 }
 
 object NodeHandle
 {
+  /** generate a [[NodeHandle]] by combining an [[InwardNodeHandle]] and an [[OutwardNodeHandle]].
+    *
+    * @param i inward node handle.
+    * @param o outward node handle.
+    * @return [[NodeHandlePair]] with inwardNode of i, outwardNode of o.
+    */
   def apply[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](i: InwardNodeHandle[DI, UI, EI, BI], o: OutwardNodeHandle[DO, UO, EO, BO]) = new NodeHandlePair(i, o)
 }
 
+/** A data structure that preserves information about the innermost and outermost Nodes in a NodeHandle. */
 class NodeHandlePair[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data]
   (inwardHandle: InwardNodeHandle[DI, UI, EI, BI], outwardHandle: OutwardNodeHandle[DO, UO, EO, BO])
   extends NodeHandle[DI, UI, EI, BI, DO, UO, EO, BO]
 {
-  val inward = inwardHandle.inward
-  val outward = outwardHandle.outward
-  def inner = inwardHandle.inner
-  def outer = outwardHandle.outer
+  /** @return [[InwardNode]] of [[inwardHandle]]. */
+  val inward: InwardNode[DI, UI, BI] = inwardHandle.inward
+
+  /** @return [[OutwardNode]] of [[outwardHandle]]. */
+  val outward: OutwardNode[DO, UO, BO] = outwardHandle.outward
+
+  /** @return the innermost [[InwardNodeImp]] of this [[NodeHandlePair]].  */
+  def inner: InwardNodeImp[DI, UI, EI, BI] = inwardHandle.inner
+
+  /** @return the outermost [[OutwardNodeImp]] of [[NodeHandlePair]]. */
+  def outer: OutwardNodeImp[DO, UO, EO, BO] = outwardHandle.outer
 }
 
+/** A handle for an [[InwardNode]], which may appear on the left side of a bind operator. */
 trait InwardNodeHandle[DI, UI, EI, BI <: Data] extends NoHandle
 {
+  /** @return [[InwardNode]] of inwardHandle. */
   def inward: InwardNode[DI, UI, BI]
+
+  /** @return [[InwardNodeImp]] of inwardHandle. */
   def inner: InwardNodeImp[DI, UI, EI, BI]
 
+  /** bind this node to an [[OutwardNodeHandle]]. */
   protected def bind[EY](h: OutwardNodeHandle[DI, UI, EY, BI], binding: NodeBinding)(implicit p: Parameters, sourceInfo: SourceInfo): Unit = inward.bind(h.outward, binding)
 
-  // connecting an input node with a full nodes => an input node
+  /** Connect an input node with a full node => an input node.
```

Comment:
I know the original comments say "input", but should this be "inward handle" or "inward node handle"?